outlines/fsm/fsm.py

```diff
@@ -37,8 +41,23 @@ class StopAtTokenFSM(FSM):
 
     def __init__(self, tokenizer: "Tokenizer", stop_token_id: int):
         self.stop_token_id = stop_token_id
-        self.vocabulary = tokenizer.vocabulary.values()
-        self.final_states = {1}
+        self.tokenizer = tokenizer
+        self.vocabulary = tokenizer.vocabulary
+        self.final_states = {2}
+        self.valid_alignment_tokens: List[int] = []
+
+    def align_prompt_tokens(self, prompt: str) -> str:
```

Comment:
`remove_boundary_token`

Can we be sure that we cannot go back more than one token? Say the prompt contains ["mo", "rn"] and the token "morning" is present in the vocabulary.

Reply:
Ah yes, right. I changed it such that it could go back as many tokens as needed